rfcs/013-gitlab-spam-checker/013-gitlab-spam-checker.md

```diff
@@ -0,0 +1,177 @@
+# Implementing a Spam Management Service for GitLab
+
+- Start Date: 2023-05-19
+- Status: Proposed
+
+## Summary
+
+This proposal aims to introduce a spam management service to mitigate spam on GitLab and improve user experience and security. This service will use machine learning models to detect and eliminate spam across the platform in real-time where possible, focusing on snippets, issues, notes, projects, and user profiles.
+
+## Problem Statement
+
+GitLab, a service on which RenkuLab relies, is not immune to the issue of spam. Spam can infiltrate various platform aspects, including snippets, issues, notes, projects, and user profiles. Spam disrupts the user experience, poses potential security risks, and damages the reputation and RenkuLab.io's SEO. GitLab lacks a robust, real-time solution to identify and eliminate spam across all attack vectors.
+
+## Key Assumptions
+
+- The proposed machine learning model will be built in consultation with the innovation team. The machine learning model's specifics are out of this RFC's scope.
+
+## Possible Solutions
+
+- A service that only checks for spam at regular intervals. This solution is less than ideal because of the delay between the spam being posted and the spam being detected and removed, during which spam could already be harmful. 
+
+- Monitor GitLab for spam manually. This solution is unsuitable because it is time-consuming.
+
+- GitLab has its [Spamcheck](https://docs.gitlab.com/ee/user/admin_area/reporting/spamcheck.html) service, which can be used to check for spam issues and snippets. This service does not check for spam in notes, projects, or user profiles. Currently, it is not possible to configure the service to delete spam automatically. Deploying GitLab's Spamcheck service also requires GitLab Enterprise Edition, which RenkuLab.io's GitLab is not currently running on.
+
+- GitLab can make use of [Akismet](https://akismet.com/), a third-party-hosted AI spam filter. However, this only checks for spam on issues and notes, needs training, and would involve handing data over to a third party in the US.
```

Comment:
Why is Akismet suitable only for issues and notes? I thought it can check any body of text? Also I thought that as a paid service it does not need to be trained.

Reply:
Partially. It doesn't need to be trained, it's learning continuously and merging all information about spam into one global model. You can contribute to its training by submitting pieces of known spam. It's limited to issues and snippets only in the way gitlab uses it, but it's made to be used for any user-entered text anywhere through its API.

Reply:
aha yes - but afaik this just trains the global model, not one for you specifically. I think it's performance is dependent on how much of the additional metadata you supply.

Reply:
I could be wrong about global vs specific - their documentation is not super obvious imho